### ISSUE TITLE
Table enhancements 

### ISSUE THREAD

**Issue by praveen-murali-ind** [bigbinary/neeto-ui#1605]
- [x] [Add functionality to reorder the columns in table](https://github.com/bigbinary/neeto-ui/issues/1575)
- [x] [Add functionality to resize the columns in table](https://github.com/bigbinary/neeto-ui/issues/1574)
- [ ] Add functionality to insert columns in between columns - feasibility study needed
- [x] [Tooltips on hovering table header](https://github.com/bigbinary/neeto-ui/issues/1588) - feature is already there, need to add an example in Storybook
- [x] Improve Table documentation
- [x] Add documentation for simple tables

cc @amaldinesh7

**Comment by Amaljith-K** [comment]
@praveen-murali-ind @amaldinesh7 We can use this issue to rebuild the table component. Let us remove the dependency on antd for the table component & let us build it on our own.

Check for the possibility of implementing features like virtualization and infinite scrolling natively.
If we could build a table component ourselves, we could eliminate a lot of boilerplates & move several repeated logic to the component itself. If we are doing that, we will need to create it as a new component & keep the current version to be backward compatible. During rollout, we can either employ someone to manually make the changes or run jscodeshift or both to start using the new table.

Whoever this task is assigned, make sure to post a comment with a detailed study of code repetition associated with the table component in neeto world. Also, include the prop schema and migration strategy so that we can have a discussion if there is any scope for improvement & edge cases. In this case, it is better to come to a clear plan before starting development.

**Comment by praveen-murali-ind** [comment]
@amaldinesh7 _a Could you please work on this?

**Comment by sumanranjanpanda** [comment]
@amaldinesh7 Any update on this? We completed 5 out of 6 tasks https://github.com/bigbinary/neeto-ui/issues/1605#issue-1609066807. 

Why is this ` Add functionality to insert columns in between columns - feasibility study needed` issue still pending?

**Comment by amaldinesh7** [comment]
I had a discussion with Amaljith regarding - https://github.com/bigbinary/neeto-ui/issues/1605#issuecomment-1530888495
That's why I kept this thread open. Created a separate issue for it https://github.com/bigbinary/neeto-ui/issues/1718.

Since we are thinking about creating new table component(moving away from AntD table), a feasibility study on `Add functionality to insert columns in between columns` on the basis of AntD table will be irrelevant. 

We can close this epic issue.

**Comment by neerajdotname** [comment]
Ref  https://github.com/bigbinary/neeto-ui/issues/1719
